### ISSUE TITLE
Five Whys: provable-contracts not enforced in apr CLI — garbage output reaches users

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#618]
### Problem Statement

`apr run` returns garbage output to users on Intel Xeon CPU (#617), and `apr parity` proves GPU/CPU compute different functions on RTX 4090 (#615) — yet both exit 0. Users receive wrong answers with no warning.

### Why 1: Why does the user see garbage?

`apr run` on Intel Xeon outputs `|் ه 发 ه 主 冗？？？？？？` instead of coherent text. The forward pass produces incorrect logits — PPL 47,350 vs expected ~19.

### Why 2: Why are the logits wrong?

`apr run` takes a different inference code path than `apr bench`. On CPU-only systems, `run` attempts GPU init (wgpu), the wgsl shader fails (#598), and the fallback to CPU uses corrupted state. `bench` skips GPU entirely and works correctly (25.7 tok/s).

### Why 3: Why doesn't the system detect and block garbage output?

No runtime contract enforcement exists in the `apr run` path. The contracts directory contains `layer-parity-v1.yaml` with cosine similarity thresholds (≥ 0.999), but these are only checked by `apr parity` — a separate diagnostic command, not an inline gate.

### Why 4: Why aren't contracts enforced inline?

The provable-contracts schema (`metadata → equations → proof_obligations → falsification_tests → kani_harnesses → qa_gate`) is designed for CI/release validation, not runtime enforcement. There is no runtime contract executor that intercepts inference output and validates against `qa_gate` pass criteria before returning results.

### Why 5: Why is there no runtime contract executor?

The architecture separates "validation tools" (`apr check`, `apr parity`, `apr qa`) from "production tools" (`apr run`, `apr serve`). The validation tools run contracts but their exit codes are broken (#601, #615, #616), and the production tools skip contracts entirely. This creates a gap where:

1. **CI gate**: Contracts exist but aren't run (no CI job runs `apr parity` with non-zero exit enforcement)
2. **Runtime gate**: No inline contract check before returning output
3. **Exit codes**: Even when contracts are run manually, failures exit 0

### Root Cause

**The provable-contracts enforcement chain is broken at every level:**

```
contracts/layer-parity-v1.yaml     ← EXISTS (correct thresholds)
    ↓
apr parity                         ← RUNS contract, DETECTS violation
    ↓                                 but EXITS 0 (#615)
apr run                            ← NEVER runs contract
    ↓                                 no inline validation
user receives garbage              ← NO GATE between inference and output
```

### Proposed Fix

1. **P0: Fix exit codes** — `apr parity`, `apr lint`, `apr eval` must return non-zero on failure (#601, #615, #616)
2. **P0: Add `--selftest` to `apr run`** — run a 1-token sanity check against known golden output before returning results. If logits diverge beyond `layer-parity-v1.yaml` thresholds, abort with error instead of returning garbage.
3. **P1: Add contract enforcement to CI** — `apr parity model.gguf` must run in CI with exit-code enforcement. Currently it would silently pass (#615).
4. **P2: Runtime contract executor** — lightweight inline check: after first forward pass, compare logit statistics against contract bounds. Fail fast if `qa_gate` criteria violated.

### Related Issues

| Issue | Role in enforcement gap |
|-------|------------------------|
| #598 | wgpu shader bug causes GPU init failure |
| #601 | lint/parity/rm exit 0 on failure |
| #607/#614 | --backend cpu flag ignored, can't force CPU path |
| #612 | GX10: CUDA works then falls to wgpu |
| #613 | Jetson: all PTX compilation fails |
| #615 | parity exits 0 when DISPROVEN |
| #616 | eval exit code regressed to 0 |
| #617 | Intel Xeon: garbage output from corrupted fallback |
